fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/security/Keys.java

```diff
@@ -23,8 +23,8 @@ public static String getEncodedKey(KeysetHandle handle) throws IOException {
     JsonObject keyData = keyObject.get("keyData").getAsJsonObject();
 
     String encoded = keyData.get("value").getAsString();
-    byte[] buf = Base64.decode(encoded, Base64.NO_WRAP);
+    byte[] buf = Base64.getUrlDecoder().decode(encoded);
 
-    return Base64.encodeToString(buf, 2, 32, Base64.NO_WRAP);
```

Comment:
Here we have 2 as an offset and 32 as the length, so we encode 32 bytes of `buf` from the position 2. You can use the flag `Base64.URL_SAFE` and keep the [android.util.Base64](https://developer.android.com/reference/android/util/Base64) method `encodeToString` or you can slice the byte array before encoding it with `getUrlEncoder().encodeToString()`